src/mumble/AudioInput.cpp

```diff
@@ -1083,8 +1083,10 @@ void AudioInput::encodeAudioFrame(AudioChunk chunk) {
 	bool bTalkingWhenMuted = false;
 	if (Global::get().s.bMute || ((Global::get().s.lmLoopMode != Settings::Local) && p && (p->bMute || p->bSuppress))
 		|| Global::get().bPushToMute || (voiceTargetID < 0)) {
-		bTalkingWhenMuted = bIsSpeech;
-		bIsSpeech         = false;
+		if (!Global::get().bPushToMute) {
+			bTalkingWhenMuted = bIsSpeech;
+		}
+		bIsSpeech = false;
```

Comment:
I think this check should not be done for setting the value of the variable because that could lead to confusion. I mean even when PTM is toggled, one _can_ be talking while being muted. And that's what this variable is supposed to represent.

Instead I would add the check further down at the same level the mute-cue setting is checked. That way we conserve the semantics of this variable while still preventing to play the mute-cue from playing if PTM is active :)

Reply:
You're right, it's clearer that way !